### PR TITLE
[Jaxrs] Add beanvalidation annotations and fix outer Enums

### PR DESCRIPTION
**Pull request by jfiala**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

* see #4091: added beanvalidation support to the language jaxrs-spec
* fixed outer enum issue as they were incorrectly generated as inline enums

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaJAXRSSpecServerCodegen.java [hunk 1]
```diff
@@ -2,27 +2,31 @@
...
 import io.swagger.util.Json;
-import org.apache.commons.io.FileUtils;
 
-public class JavaJAXRSSpecServerCodegen extends AbstractJavaJAXRSServerCodegen
-{	
-	public JavaJAXRSSpecServerCodegen()
-	{
+public class JavaJAXRSSpecServerCodegen extends AbstractJavaJAXRSServerCodegen implements BeanValidationFeatures
+{    
+    protected boolean useBeanValidation = true;
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaJAXRSSpecServerCodegen.java]: I suggest we keep `useBeanValidation` to false similar to what we've done for other Java-related generator: https://github.com/swagger-api/swagger-codegen/search?l=Java&q=useBeanValidation&utf8=%E2%9C%93

Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaJAXRSSpecServerCodegen.java]: @jfiala I'll think about this more and get back to you.

**Comment by wing328** [comment]
@jfiala the Travis CI failed with the following error message: 
```
The command "set -e" exited with 0.
$ /bin/bash ./bin/utils/detect_carriage_return.sh
modules/swagger-codegen/src/main/resources/JavaJaxRS/spec/generatedAnnotation.mustache
modules/swagger-codegen/src/main/resources/JavaJaxRS/spec/beanValidation.mustache
Templates contain carriage return '/r'. Please remove it and try again.
```
Please take a look.

**Comment by jfiala** [comment]
@wing328 I changed my git settings (on Windows) to git config --global core.autocrlf input and then did a reset for the branch as described here (https://help.github.com/articles/dealing-with-line-endings/#refreshing-a-repository-after-changing-line-endings) and checked in with corrected lf settings.
Is the recommended setting for autocrlf documented somewhere?

**Comment by jfiala** [comment]
@wing328 the crlf are fixed now (one build is still failing due to an error at the Java library ok-http-gson, but I didn't change anything there).

**Comment by wing328** [comment]
> Is the recommended setting for autocrlf documented somewhere?

It's not documented in our wiki at the moment. I'll put it in the contributing guideline shortlly.

Thanks for the suggestion 👍

**Comment by wing328** [comment]
Documented in https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md#tips

**Comment by jfiala** [comment]
@wing328 is there already an ETA for the next release 2.2.2?
I'd like to further work on the Jaxrs-languages (interface etc.), but I need my Beanvalidation-PRs to first get pulled in.
Should I add additional branches starting from the jaxrs-branches in my forked repo? Or is it better to wait for my PRs to get pulled in so I can restart from the (upstream) master?

**Comment by wing328** [comment]
@jfiala let me try to review later today and merge this in if there's no feedback from me or anyone from the community.

ETA for 2.2.2 is Jan/Feb 2017

**Comment by wing328** [comment]
@jfiala I ran some tests and didn't find any issue. Thanks for the enhancements.

**Comment by jfiala** [comment]
@wing328 great, thx can you pls also take a look at the other beanvalidation PR's so we get them all in?

**Comment by wing328** [comment]
@jfiala yup, merged a few already. Would need your help to rebase one of them on the latest master.

**Comment by jfiala** [comment]
@wing328 thx alot, I already noticed, I'll rebase the spring-branch and then let you know.

**Pull request merged**